datahub-graphql-api/src/main/java/com/linkedin/datahub/graphql/resolvers/AuthorizedResolver.java

```diff
@@ -0,0 +1,29 @@
+package com.linkedin.datahub.graphql.resolvers;
+
+import com.linkedin.datahub.graphql.exception.AuthorizationError;
+import graphql.schema.DataFetchingEnvironment;
+
+/**
+ * Abstract base class that should be leveraged for any operation that should be protected by authorization.
+ */
+public abstract class AuthorizedResolver<T> extends AuthenticatedResolver<T> {
+
+    /**
+     * Abstract method to be overridden by subclasses to implement a authorized resolver.
+     */
+    protected abstract boolean isAuthorized(DataFetchingEnvironment environment) throws Exception;
+
+    /**
+     * Abstract method to be overridden by subclasses to implement a secured resolver.
+     */
+    protected abstract T authorizedGet(DataFetchingEnvironment environment) throws Exception;
+
+    @Override
+    final public T authenticatedGet(DataFetchingEnvironment environment) throws Exception {
+        if (isAuthorized((environment))) {
+            return authorizedGet(environment);
+        }
+        throw new AuthorizationError("Failed to authorize the current user.");
```

Comment:
You're correct in understanding that this exception will not be thrown in a case of "something went wrong". This will be thrown when an authorization check fails, meaning that the current authenticated user is not authorized to perform a particular action. 

`UserNotAuthorizedException` is overly verbose. `AuthorizationError` should get the point across just fine. To clarify the domain-specific authorization scenario, perhaps the isAuthorized method can return a message to be included in the error. Thoughts?

Reply:
Introduced a small `AuthorizationResult` container object that allows the overrider to provide an optional message that will be carried with the exception.